### ISSUE TITLE
Build problem, Ubuntu 19.10

### ISSUE THREAD

**Issue by jfacemyer** [aramg/droidcam#69]
I get errors using either install or install-dkms on Ubuntu 19.10:

install:

`Building v4l2loopback-dc.ko
make: Entering directory '/home/facemyer/bin/droidcam/linux/v4l2loopback'
make -C /lib/modules/5.3.0-55-generic/build M=/home/facemyer/bin/droidcam/linux/v4l2loopback modules
make[1]: Entering directory '/usr/src/linux-headers-5.3.0-55-generic'
  Building modules, stage 2.
  MODPOST 1 modules
make[1]: Leaving directory '/usr/src/linux-headers-5.3.0-55-generic'
make: Leaving directory '/home/facemyer/bin/droidcam/linux/v4l2loopback'
./install: line 23: mokutil: command not found
Adding uninstall script
Moving driver and executable to system folders
+ cp v4l2loopback/v4l2loopback-dc.ko /lib/modules/5.3.0-55-generic/kernel/drivers/media/video/
+ cp droidcam /usr/bin/
cp: cannot stat 'droidcam': No such file or directory`

install-dkms:

`Adding uninstall script in '/opt/droidcam-uninstall'
droidcam was not previously built. Installation skip.
droidcam-cli was not previously built. Installation skip.
Module v4l2loopback_dc/0.0.1 already built for kernel 5.3.0-55-generic/4
Module v4l2loopback_dc/0.0.1 already installed on kernel 5.3.0-55-generic/x86_64
Using /etc/dkms/template-dkms-mkdeb
copying template...
modifying debian/changelog...
modifying debian/compat...
modifying debian/control...
modifying debian/copyright...
modifying debian/dirs...
modifying debian/postinst...
modifying debian/prerm...
modifying debian/README.Debian...
modifying debian/rules...
copying legacy postinstall template...
Copying source tree...
Gathering binaries...Marking modules for 5.3.0-55-generic (x86_64) for archiving...

Creating tarball structure to specifically accomodate binaries.

Tarball location: /var/lib/dkms/v4l2loopback_dc/0.0.1/tarball/v4l2loopback_dc-0.0.1.dkms.tar.gz

DKMS: mktarball completed.

Copying DKMS tarball into DKMS tree...
Building binary package...dpkg-buildpackage: warning: using a gain-root-command while being root
 dpkg-source --before-build .
 fakeroot debian/rules clean
dh_clean: Compatibility levels before 9 are deprecated (level 7 in use)
 debian/rules build
 fakeroot debian/rules binary
dh_installdirs: Compatibility levels before 9 are deprecated (level 7 in use)
dh_strip: Compatibility levels before 9 are deprecated (level 7 in use)
dh_compress: Compatibility levels before 9 are deprecated (level 7 in use)
dh_installdeb: Compatibility levels before 9 are deprecated (level 7 in use)
dh_shlibdeps: Compatibility levels before 9 are deprecated (level 7 in use)
 dpkg-genbuildinfo --build=binary
 dpkg-genchanges --build=binary >../v4l2loopback-dc-dkms_0.0.1_amd64.changes
dpkg-genchanges: info: binary-only upload (no source code included)
 dpkg-source --after-build .

DKMS: mkdeb completed.
Moving built files to /var/lib/dkms/v4l2loopback_dc/0.0.1/deb...
Cleaning up temporary files...
modprobe: ERROR: could not insert 'v4l2loopback_dc': Exec format error`

Am I doing something wrong?

**Comment by jfacemyer** [comment]
FYI, I'm trying to get dkms to work so I don't have to use the install script every time I reboot. That's why I'm trying this rather than the official working package (which does work for me).

**Comment by aramg** [comment]
You can see, `./install: line 23: mokutil: command not found`
Try `apt-get install mokutil`
If the regular install works, then you can try dkms

**Comment by jfacemyer** [comment]
Thanks for the reply!

I ignored that because the published tarball builds and works with the same error. (Should this be added as a dep on the readme?)

So, I updated from git just now, installed mokutils and all deps on the readme (looks like they're for the gtk3 update?)

Same thing - looks like the driver gets built, but not the executables? No errors except that it can't stat droidcam (and indeed, it doesn't exist). And when I do the dkms install, it looks fine, except it errors out on modprobe with "exec format error", so looks like that's not building correctly either.

`sudo ./install
Webcam parameters: '640' and '480'
Building v4l2loopback-dc.ko
make: Entering directory '/home/facemyer/bin/droidcam-git/linux/v4l2loopback'
make -C /lib/modules/5.3.0-55-generic/build M=/home/facemyer/bin/droidcam-git/linux/v4l2loopback modules
make[1]: Entering directory '/usr/src/linux-headers-5.3.0-55-generic'
  CC [M]  /home/facemyer/bin/droidcam-git/linux/v4l2loopback/v4l2loopback-dc.o
  Building modules, stage 2.
  MODPOST 1 modules
  CC      /home/facemyer/bin/droidcam-git/linux/v4l2loopback/v4l2loopback-dc.mod.o
  LD [M]  /home/facemyer/bin/droidcam-git/linux/v4l2loopback/v4l2loopback-dc.ko
make[1]: Leaving directory '/usr/src/linux-headers-5.3.0-55-generic'
make: Leaving directory '/home/facemyer/bin/droidcam-git/linux/v4l2loopback'
EFI variables are not supported on this system
Adding uninstall script
Moving driver and executable to system folders
+ cp v4l2loopback/v4l2loopback-dc.ko /lib/modules/5.3.0-55-generic/kernel/drivers/media/video/
+ cp droidcam /usr/bin/
cp: cannot stat 'droidcam': No such file or directory
`
`sudo ./install-dkms 
Webcam parameters: '640' and '480'
Adding uninstall script in '/opt/droidcam-uninstall'
droidcam was not previously built. Installation skip.
droidcam-cli was not previously built. Installation skip.
Module v4l2loopback_dc/0.0.1 already built for kernel 5.3.0-55-generic/4
Module v4l2loopback_dc/0.0.1 already installed on kernel 5.3.0-55-generic/x86_64
Using /etc/dkms/template-dkms-mkdeb
copying template...
modifying debian/changelog...
modifying debian/compat...
modifying debian/control...
modifying debian/copyright...
modifying debian/dirs...
modifying debian/postinst...
modifying debian/prerm...
modifying debian/README.Debian...
modifying debian/rules...
copying legacy postinstall template...
Copying source tree...
Gathering binaries...Marking modules for 5.3.0-55-generic (x86_64) for archiving...

Creating tarball structure to specifically accomodate binaries.

Tarball location: /var/lib/dkms/v4l2loopback_dc/0.0.1/tarball/v4l2loopback_dc-0.0.1.dkms.tar.gz


DKMS: mktarball completed.

Copying DKMS tarball into DKMS tree...
Building binary package...dpkg-buildpackage: warning: using a gain-root-command while being root
 dpkg-source --before-build .
 fakeroot debian/rules clean
dh_clean: Compatibility levels before 9 are deprecated (level 7 in use)
 debian/rules build
 fakeroot debian/rules binary
dh_installdirs: Compatibility levels before 9 are deprecated (level 7 in use)
dh_strip: Compatibility levels before 9 are deprecated (level 7 in use)
dh_compress: Compatibility levels before 9 are deprecated (level 7 in use)
dh_installdeb: Compatibility levels before 9 are deprecated (level 7 in use)
dh_shlibdeps: Compatibility levels before 9 are deprecated (level 7 in use)
 dpkg-genbuildinfo --build=binary
 dpkg-genchanges --build=binary >../v4l2loopback-dc-dkms_0.0.1_amd64.changes
dpkg-genchanges: info: binary-only upload (no source code included)
 dpkg-source --after-build .


DKMS: mkdeb completed.
Moving built files to /var/lib/dkms/v4l2loopback_dc/0.0.1/deb...
Cleaning up temporary files...
modprobe: ERROR: could not insert 'v4l2loopback_dc': Exec format error
`

Thanks!
